### PR TITLE
V17.0 increment version

### PR DESCRIPTION
**Pull request by albi3ro**
This PR increments the version variable, updates the changelog, adds a gitignore entry to a build artifacts folder, and corrects the test building command on the readme.

### CODE REVIEW

**Review by trbromley — APPROVED** [review]
Looks good! Why are we including abf7b99 and 14ea923e85bffae9bc552415c293f6385ff5bf6d which have happened post feature freeze? 14ea923e85bffae9bc552415c293f6385ff5bf6d is a bug fix so that's fine, but abf7b99 is a new feature and probably skippable (if we're adhering strictly to feature freeze).

**Review by antalszava — APPROVED** [review]
Looks good to me! :tada: Maybe indeed worth leaving the post-feature freeze commits out, as Tom pointed out. That just helps with reflecting the state of the release candidate.

**Comment by antalszava** [comment]
[ch7894]

File: pennylane_lightning/_version.py [hunk 1]
```diff
@@ -16,4 +16,4 @@
    Version number (major.minor.patch[-label])
 """
 
-__version__ = "0.16.0-dev"
+__version__ = "0.17.0"
```
[the remark this answers is not in the document]
Comment (antalszava) [reply pennylane_lightning/_version.py:19]: We'd usually do that if there's a feature that lightning depends on after v0.15.0. That is: we'd pin to the latest released version of PennyLane that the plugin is still fully functional with.

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -1,48 +1,64 @@
...
+  [(#111)](https://github.com/PennyLaneAI/pennylane-lightning/pull/111)
+
+* Adds CMake build system support.
+  [(#104)](https://github.com/PennyLaneAI/pennylane-lightning/pull/104)
+
 
 ### Breaking changes
 
+* Removes support for Python 3.6.
+  [#127](https://github.com/PennyLaneAI/pennylane-lightning/pull/127)
```
Comment (antalszava) [inline .github/CHANGELOG.md:42]:
```suggestion
  [(#127)](https://github.com/PennyLaneAI/pennylane-lightning/pull/127)
```

**Pull request merged**